xapian-letor/api/letor_internal.cc

```diff
@@ -204,6 +210,68 @@ static void read_problem(const char *filename) {
 	while (1) {
 	    idx = strtok(NULL, ":");
 	    val = strtok(NULL, " \t");
+    std::vector<FeatureVector> fvv;
+    string lastqid;
+    int k =0;
+    while (train_file.peek() != EOF) {
+        k++;
+        FeatureVector fv;//new a fv
+
+        double label;//read label
+        train_file >> label;
+        fv.set_label(label);
+        train_file.ignore();
+
+        string qid;//read qid
+        train_file >> qid;
+
+        qid = qid.substr(qid.find(":")+1,qid.length());
+
+        if (lastqid==""){
+            lastqid = qid;
+        }
+        //std::cout << fv.did << endl;
+
+        std::map<int,double> fvals;//read features
+        for(int i = 1; i < 20; ++i){
+            train_file.ignore();
+            int feature_index;
+            double feature_value;
+            train_file >> feature_index;
+            train_file.ignore();
+            train_file >> feature_value;
+            //std::cout<<feature_index<<" "<<feature_value<<endl;
+            fvals[feature_index] = feature_value;
```

Comment:
If we go with the Enum style feature references, which I strongly believe soon we should, it should be resolved with the reverse map/dictionary of features with ids.